src/command/render/latexmk/texlive.ts

```diff
@@ -280,21 +281,39 @@ async function verifyPackageInstalled(
   return result.stdout?.trim() === pkg;
 }
 
-function tlmgrCommand(
+async function tlmgrCommand(
   cmd: string,
   args: string[],
   _quiet?: boolean,
 ) {
+  let cmdExec: string[];
+  let tempFile: string | undefined;
+  if (Deno.build.os === "windows") {
+    // quote args on Windows
+    args = args.map((a) => `"${a}"`);
+    // writing to a file to exectute to avoid quoting issue with Deno
+    // https://github.com/quarto-dev/quarto-cli/issues/336
+    const lines = ["tlmgr", cmd, ...args];
+    tempFile = Deno.makeTempFileSync(
```

Comment:
I understand. I'll modify accordingly

Reply:
Totally agree re function argument, this is nicer!

You could make that `let tempFile` be a const if you move the creation of the temp file right there. Also means your finally wouldn't have to be conditional...

Reply:
> You could make that let tempFile be a const if you move the creation of the temp file right there. Also means your finally wouldn't have to be conditional...

I wanted to do that but that is the part that I don't get 😅 
If I do that, I have warning from Typescript in VSCODE
![image](https://user-images.githubusercontent.com/6791940/154310342-640f0540-8b00-471e-a995-a9f19eb2bd0d.png)

and then error 
````
error: TS2304 [ERROR]: Cannot find name 'tempFile'.
      removeIfExists(tempFile);
                     ~~~~~~~~
    at file:///C:/Users/chris/DOCUME~1/DEV_R/QUARTO~2/src/command/render/latexmk/texlive.ts:306:22
````

That is why I did defined it outside of the try catch. 

am I missing a typescript special syntax ?

Reply:
I think you want:

```typescript
if (Deno.build.os === "windows") {
     // On Windows writing the command to a file to avoid quoting issue with Deno
     // https://github.com/quarto-dev/quarto-cli/issues/336
     const tempFile = Deno.makeTempFileSync(
         { prefix: "tlmgr-cmd", suffix: ".bat" },
       );
     try {
       // Quoting every argument for CMD
       args = args.map((a) => `"${a}"`);
       const lines = ["tlmgr", cmd, ...args];
```

that should make it visible to finally (and if creating the temp file throws, no need to clean it up so safe outside the try)

Reply:
Oh I see. I thought it would need to be in the try, but you're right. No need to clean, and the other try in the other function  will catch that.

Reply:
I have put out the variables in [62ceb48](https://github.com/quarto-dev/quarto-cli/pull/340/commits/62ceb4856953d890ecce17f27df9e1b26eeb03e3)